internal/services/integritymonitor/integritymonitor.go

```diff
@@ -115,102 +82,50 @@ func SetupIntegrity(ctx context.Context, monitoringDirectory string, log *logrus
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	errC := make(chan error)
+	defer close(errC)
+
+	kubeData, err := kubeClient.GetKubeData()
 	if err != nil {
-		m.logger.WithError(err).Error("get data from k8s API")
 		return err
 	}
-	fileHashesDto, err := m.repository.GetHashData(
-		m.monitoringDirectory,
-		algName,
-		k8sData.DeploymentData,
-	)
+	deploymentDta, err := kubeClient.GetDataFromDeployment(kubeData)
 	if err != nil {
-		return fmt.Errorf("failed get hash data: %w", err)
-	}
-
-	referenceHashes := make(map[string]*models.HashDataFromDB, len(fileHashesDto))
-
-	for _, fh := range fileHashesDto {
-		referenceHashes[fh.FullFilePath] = fh
-	}
-
-	for _, fh := range fileHashes {
-		if fhdto, ok := referenceHashes[fh.Path]; ok {
-			if fhdto.Hash != fh.Hash {
-				m.integrityCheckFailed(
-					ErrIntegrityFileMismatch,
-					fh.Path,
-					k8sData.KubeData,
-					k8sData.DeploymentData,
-				)
-				return ErrIntegrityFileMismatch
-			}
-			delete(referenceHashes, fh.Path)
-		} else {
-			m.integrityCheckFailed(ErrIntegrityNewFileFoud, fh.Path, k8sData.KubeData, k8sData.DeploymentData)
-			return ErrIntegrityNewFileFoud
-		}
-	}
-
-	if len(referenceHashes) > 0 {
-		for path := range referenceHashes {
-			m.integrityCheckFailed(
-				ErrIntegrityFileDeleted,
-				path,
-				k8sData.KubeData,
-				k8sData.DeploymentData,
-			)
-			return ErrIntegrityFileDeleted
-		}
+		return err
 	}
 
-	m.logger.Debug("end check integrity")
-	return err
-}
-
-func (m *IntegrityMonitor) integrityCheckFailed(
-	err error,
-	path string,
-	kubeData *models.KubeData,
-	deploymentData *models.DeploymentData,
-) {
-	switch err {
-	case ErrIntegrityFileMismatch:
-		m.logger.WithField("path", path).Warn("file content missmatch")
-	case ErrIntegrityNewFileFoud:
-		m.logger.WithField("path", path).Warn("new file found")
-	case ErrIntegrityFileDeleted:
-		m.logger.WithField("path", path).Warn("file deleted")
-	}
-	if m.alertSender != nil {
-		err := m.alertSender.Send(alerts.Alert{
-			Time:    time.Now(),
-			Message: fmt.Sprintf("Restart deployment %v", deploymentData.NameDeployment),
-			Reason:  "mismatch file content",
-			Path:    path,
-		})
-		if err != nil {
-			m.logger.WithError(err).Error("Failed send alert")
+	log.Trace("calculate & save hashes...")
+	for {
+		select {
+		case <-ctx.Done():
+			log.Error(ctx.Err())
+			return ctx.Err()
+		case countHashes := <-compareHashes(
+			ctx,
+			worker.WorkersPool(
+				viper.GetInt("count-workers"),
+				walker.ChanWalkDir(ctx, monitoringDirectory, log),
+				worker.NewWorker(ctx, viper.GetString("algorithm"), log),
+			),
+			repo,
+			monitoringDirectory,
+			viper.GetString("algorithm"),
+			deploymentDta,
+			errC,
+		):
+			log.WithField("countHashes", countHashes).Info("hashes compared successfully")
+			log.Debug("end check integrity")
+			return nil
+		case err := <-errC:
+			integrityCheckFailed(err, log, alertSender, kubeClient)
```

Comment:
The `kubeClient` arg is not necessary to path since we already have collected `deploymentData`

Reply:
integrityCheckFailed require kubeClient to restart pod